src/Cache/ConnectionCache.php

```diff
@@ -0,0 +1,190 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Yiisoft\Db\Cache;
+
+use Psr\SimpleCache\CacheInterface;
+use Yiisoft\Cache\Dependency\Dependency;
+
+use function array_pop;
+use function end;
+use function is_array;
+
+final class ConnectionCache
+{
+    private CacheInterface $schemaCache;
+    private bool $enableSchemaCache = true;
+    private int $schemaCacheDuration = 3600;
+    private array $schemaCacheExclude = [];
+    private bool $enableQueryCache = true;
+    public array $queryCacheInfo = [];
+    private int $queryCacheDuration = 3600;
+
+    public function __construct(CacheInterface $schemaCache)
+    {
+        $this->schemaCache = $schemaCache;
+    }
+
+    public function getCacheKey(array $key): string
+    {
+        $jsonKey = json_encode($key, JSON_THROW_ON_ERROR);
+
+        return md5($jsonKey);
+    }
+
+    public function getQueryCacheDuration(): ?int
+    {
+        return $this->queryCacheDuration;
+    }
+
+    public function getQueryCacheInfo(): array
+    {
+        return $this->queryCacheInfo;
+    }
+
+    public function getSchemaCache(): CacheInterface
+    {
+        return $this->schemaCache;
+    }
+
+    public function getSchemaCacheDuration(): int
+    {
+        return $this->schemaCacheDuration;
+    }
+
+    public function getSchemaCacheExclude(): array
+    {
+        return $this->schemaCacheExclude;
+    }
+
+    public function isQueryCacheEnabled(): bool
+    {
+        return $this->enableQueryCache;
+    }
+
+    public function isSchemaCacheEnabled(): bool
+    {
+        return $this->enableSchemaCache;
+    }
+
+    /**
+     * Returns the current query cache information.
+     *
+     * This method is used internally by {@see Command}.
+     *
+     * @param int|null $duration the preferred caching duration. If null, it will be ignored.
+     * @param Dependency|null $dependency the preferred caching dependency. If null, it will be
+     * ignored.
+     *
+     * @return array|null the current query cache information, or null if query cache is not enabled.
+     */
+    public function queryCacheInfo(?int $duration, Dependency $dependency = null): ?array
+    {
+        $result = null;
+
+        if ($this->enableQueryCache) {
+            $info = end($this->queryCacheInfo);
+
+            if (is_array($info)) {
+                if ($duration === null) {
+                    $duration = $info[0];
+                }
+
+                if ($dependency === null) {
+                    $dependency = $info[1];
+                }
+            }
+
+            if ($duration === 0 || $duration > 0) {
+                if ($this->schemaCache instanceof CacheInterface) {
```

Comment:
Either it's a bad naming (i.e. schema cache should be named just `cache`) or the idea was to split schema cache and query cache. Which is true?

Reply:
Since yii2 both caches are separated into SchemaCache and QueryCache, but in reality it is the same cache for both, with their different configurations.

Reply:
Done.